source/common/tracing/zipkin_tracer_impl.cc

```diff
@@ -0,0 +1,194 @@
+  if (!cluster) {
+    throw EnvoyException(fmt::format("{} collector cluster is not defined on cluster manager level",
+                                     config.getString("collector_cluster")));
+  }
+  cluster_ = cluster->info();
+
+  if (cluster_->features() & Upstream::ClusterInfo::Features::HTTP2) {
+    throw EnvoyException(
+        fmt::format("Zipkin collector service (cluster {}) can be accessed over http1.1 only",
+                    cluster_->name()));
+  }
+
+  const std::string collector_endpoint = config.getString("collector_endpoint");
+
+  tls_.set(tls_slot_, [this, collector_endpoint, &random_generator](Event::Dispatcher& dispatcher)
+                          -> ThreadLocal::ThreadLocalObjectSharedPtr {
+                            Zipkin::Tracer tracer(local_info_.clusterName(), local_info_.address());
+                            tracer.setReporter(ZipkinReporter::NewInstance(
+                                std::ref(*this), std::ref(dispatcher), collector_endpoint));
+                            Runtime::RandomGeneratorPtr rand_ptr(&random_generator);
+                            tracer.setRandomGenerator(std::move(rand_ptr));
+                            return ThreadLocal::ThreadLocalObjectSharedPtr{
+                                new TlsZipkinTracer(std::move(tracer), *this)};
+                          });
+}
+
+SpanPtr ZipkinDriver::startSpan(Http::HeaderMap& request_headers, const std::string&,
+                                SystemTime start_time) {
+  Zipkin::Tracer& tracer = *tls_.getTyped<TlsZipkinTracer>(tls_slot_).tracer_;
+  ZipkinSpanPtr active_span;
+  Zipkin::SpanPtr new_zipkin_span;
+
+  if (request_headers.OtSpanContext()) {
+    // Get the open-tracing span context.
+    // This header contains a span's parent-child relationships set by the downstream Envoy.
+    // The context built from this header allows the Zipkin tracer to
+    // properly set the span id and the parent span id.
+    Zipkin::SpanContext context;
+
+    context.populateFromString(request_headers.OtSpanContext()->value().c_str());
+
+    // Create either a child or a shared-context Zipkin span.
+    new_zipkin_span =
+        tracer.startSpan(request_headers.Host()->value().c_str(), start_time, context);
+  } else {
+    // Create a root Zipkin span.
+    new_zipkin_span = tracer.startSpan(request_headers.Host()->value().c_str(), start_time);
+  }
+
+  // Set the trace-id and span-id headers properly, based on the newly-created span structure.
+  request_headers.insertXB3TraceId().value(new_zipkin_span->traceIdAsHexString());
+  request_headers.insertXB3SpanId().value(new_zipkin_span->idAsHexString());
+
+  // Set the parent-span header properly, based on the newly-created span structure.
+  if (new_zipkin_span->isSetParentId()) {
+    request_headers.insertXB3ParentSpanId().value(new_zipkin_span->parentIdAsHexString());
+  }
+
+  // Set the sampled header.
+  request_headers.insertXB3Sampled().value(std::string(("1")));
```

Comment:
can you create static const string with the name ALWAYS_SAMPLE or something instead of std::string(("1"))